org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/DateRangeEditor.java

```diff
@@ -0,0 +1,142 @@
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the        *
+* GNU General Public License for more details.                        *
+*                                                                     *
+* You should have received a copy of the GNU General Public License   *
+* along with this program; if not, write to the Free Software         *
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,          *
+* MA 02110-1301, USA.                                                 *
+*                                                                     *
+* Contributors:                                                       *
+* - Peter Takacs, Cloudempiere                                        *
+**********************************************************************/
+package org.adempiere.webui.window;
+
+import org.adempiere.webui.LayoutUtils;
+import org.adempiere.webui.component.Button;
+import org.adempiere.webui.component.Textbox;
+import org.adempiere.webui.component.ToolBarButton;
+import org.adempiere.webui.component.ZkCssHelper;
+import org.adempiere.webui.editor.WEditor;
+import org.adempiere.webui.event.ValueChangeEvent;
+import org.adempiere.webui.event.ValueChangeListener;
+import org.adempiere.webui.theme.ThemeManager;
+import org.zkoss.zk.ui.event.Events;
+import org.zkoss.zul.Div;
+
+/**
+*
+* @author Peter Takacs, Cloudempiere
+*
+*/
+public class DateRangeEditor extends Div implements ValueChangeListener {
+
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 3329360781729704243L;
+
+	private static final String IMAGES_CONTEXT_CALENDAR_PNG = "images/Calendar16.png";
+
+	private Textbox textbox;
+	ToolBarButton popupBtn;
+	private DateRangePicker popup;
+	private boolean isProcessParameter;
+	private WEditor editor;
+	private WEditor editor2;
+
+	/**
+	 * Constructor
+	 */
+	public DateRangeEditor(WEditor editor, WEditor editor2, boolean isProcessParameter) {
+		super();
+		this.isProcessParameter = isProcessParameter;
+		this.editor = editor;
+		this.editor2 = editor2;
+		init();
+	}
+
+	private void init() {
+
+		this.setWidth("100%");
+		ZkCssHelper.appendStyle(this, "display: inline-block; position: relative;");
+		textbox = new Textbox();
+		textbox.setStyle("width: 100%; background: white !important");
+		textbox.setReadonly(true);
+		this.appendChild(textbox);
+
+		popupBtn = new ToolBarButton();
+		popupBtn.addSclass("date-picker-calendar-button");
+		if (ThemeManager.isUseFontIconForImage())
+			popupBtn.setIconSclass("z-icon-Calendar");
+		else
+			popupBtn.setImage(ThemeManager.getThemeResource(IMAGES_CONTEXT_CALENDAR_PNG));
+		this.appendChild(popupBtn);
+
+		popup = new DateRangePicker(editor, editor2);
+		popupBtn.setTooltip(popup);
+		popupBtn.addEventListener(Events.ON_CLICK, event -> {
+			popup.setPage(popupBtn.getPage());
+			popup.open(popupBtn, "after_center");
+			LayoutUtils.autoDetachOnClose(popup);
+		});
+
+		Button btn = popup.getUpdateButton();
```

Comment:
the more natural way of doing this is to register this class as listener or callback to DateRangePicker and DateRangePicker will invoke the register listener or callback when it is dismiss by user. The current approach is not good since the UI design of DateRangePicker can change and DateRangePicker can be dismissed by user through other means (for e.g, keyboard shortcut).